mod-record-specifications-validator/src/test/java/org/folio/rspec/validation/validator/marc/impl/InvalidLccnSubfieldRuleValidatorTest.java

```diff
@@ -0,0 +1,109 @@
+import org.folio.rspec.validation.validator.marc.model.Reference;
+import org.folio.spring.testing.type.UnitTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.util.CollectionUtils;
+
+@UnitTest
+@ExtendWith(MockitoExtension.class)
+public class InvalidLccnSubfieldRuleValidatorTest {
+
+  @Mock
+  private TranslationProvider translationProvider;
+  @InjectMocks
+  private InvalidLccnSubfieldRuleValidator validator;
+
+  @Test
+  void validate_whenNon010Field_shouldReturnEmptyList() {
+    var errors = validator.validate(List.of(), new SpecificationFieldDto().tag("tag"));
+
+    assertTrue(CollectionUtils.isEmpty(errors));
+  }
+
+  @Test
+  void validate_whenLccnSubfieldNotExist_shouldReturnEmptyList() {
+    var subfield = new MarcSubfield(Reference.forSubfield(Reference.forTag("010"), 'z'), "12345");
+
+    var errors = validator.validate(List.of(subfield), new SpecificationFieldDto().tag("010"));
+
+    assertTrue(CollectionUtils.isEmpty(errors));
+  }
+
+  @ParameterizedTest
+  @MethodSource("validLccn")
+  void validate_whenValidlccn_shouldReturnEmptyList(String lccn) {
+    var subfield = new MarcSubfield(Reference.forSubfield(Reference.forTag("010"), 'a'), lccn);
+
+    var errors = validator.validate(List.of(subfield), new SpecificationFieldDto().tag("010"));
+
+    assertTrue(CollectionUtils.isEmpty(errors));
+  }
+
+  @ParameterizedTest
+  @MethodSource("invalidLccn")
+  void validate_whenInvalidlccn_shouldReturnValidationError(String lccn) {
+    when(translationProvider.format(anyString(), anyString(), anyString())).thenReturn("message");
+    var subfield = new MarcSubfield(Reference.forSubfield(Reference.forTag("010"), 'a'), lccn);
+    var specification = new SpecificationFieldDto().tag("010")
+      .subfields(List.of(new SubfieldDto().code("a")));
+
+    var errors = validator.validate(List.of(subfield), specification);
+
+    assertEquals(1, errors.size());
+    ValidationError error = errors.get(0);
+    assertEquals(validator.definitionType(), error.getDefinitionType());
+    assertEquals(validator.severity(), error.getSeverity());
+    assertEquals(validator.supportedRule().getCode(), error.getRuleCode());
+    assertEquals("message", error.getMessage());
+  }
+
+  public static Stream<Arguments> validLccn() {
+    return Stream.of(
+      // structure A
+      arguments("12345678"),
+      arguments("n12345678"),
+      arguments("nn12345678"),
+      arguments("nnn12345678"),
+
+      // structure B
+      arguments("0123456789"),
+      arguments("n0123456781"),
+      arguments("nn0123456789"),
+      arguments("nnn0123456789"));
```

Comment:
Based on structure B format there should be 2 character prefix (alpha or whitespace)